gst/edge/edge_sink.c

```diff
@@ -311,6 +338,17 @@ gst_edgesink_render (GstBaseSink * basesink, GstBuffer * buffer)
   GstMemory *mem[NNS_TENSOR_SIZE_LIMIT];
   GstMapInfo map[NNS_TENSOR_SIZE_LIMIT];
 
+  if (self->wait_connection) {
+    guint64 counter = 0, timeout = self->connection_timeout;
+    if (0 == timeout)
+      timeout = G_MAXUINT64;
+
+    while ((timeout / 10) > counter++ &&
+        NNS_EDGE_ERROR_NONE != nns_edge_is_connected (self->edge_h)) {
+      g_usleep (10000);
+    }
+  }
+
```

Comment:
How about moving this code to ```gst_edgesink_start (line 324)```?

Reply:
Thanks. It would be better to move as you suggested. Updated.